nucypher/blockchain/eth/token.py

```diff
@@ -883,3 +889,10 @@ def __read_stakes(self) -> None:
 
         # Record most recent cache update
         self.__updated = maya.now()
+
+    @property
+    def has_active_substakes(self) -> bool:
+        current_period = self.staking_agent.get_current_period()
+        for stake in self.data:
+            if not stake.status(current_period=current_period).is_child(Stake.Status.INACTIVE):
```

Comment:
You don't need to pass `current_period` here, if `status()` doesn't receive a current_period value it derives it - not a big deal but would make this method look even cleaner.

(From `status()` method)
```python
current_period = current_period or self.staking_agent.get_current_period()  # TODO #1514 this is online only.
```

Reply:
I should have left the comment there: I did it intentionally to avoid multiple calls to contract. I thought it was neat, but maybe it's a bit pendantic.

Reply:
`get_current_period()` would only be called once either way no? Either in this function, `has_active_substakes` or in `status()` - my comment was purely a simplicity change, not a functional change.

Reply:
Yes, it would be called once per stake in a loop. So I figured I could just call it once outside the loop.

Reply:
🤦 - You are totally right - ignore me.